addons/api/addon/serializers/host-catalog.js

```diff
@@ -4,30 +4,51 @@
  */
 
 import ApplicationSerializer from './application';
+import {
+  TYPE_HOST_CATALOG_PLUGIN_AWS,
+  TYPE_HOST_CATALOG_PLUGIN_AZURE,
+} from '../models/host-catalog';
 
-const fieldsByType = {
-  aws: [
-    'disable_credential_rotation',
-    'region',
+const azureFields = [
+  'disable_credential_rotation',
+  'tenant_id',
+  'client_id',
+  'subscription_id',
+  'secret_id',
+  'secret_value',
+];
+
+const AWSfieldsWithCredentialType = {
+  static: [
     'access_key_id',
     'secret_access_key',
+    'region',
+    'disable_credential_rotation',
     'worker_filter',
   ],
-  azure: [
+  dynamic: [
+    'role_arn',
+    'role_external_id',
+    'role_session_name',
+    'role_tags',
+    'region',
     'disable_credential_rotation',
-    'tenant_id',
-    'client_id',
-    'subscription_id',
-    'secret_id',
-    'secret_value',
+    'worker_filter',
   ],
 };
 
 export default class HostCatalogSerializer extends ApplicationSerializer {
-  // =methods
   serialize(snapshot) {
+    const { compositeType } = snapshot.record;
     const serialized = super.serialize(...arguments);
-    switch (snapshot.record.compositeType) {
+
+    // This is a hack to remove the worker_filter field from the serialized data if the compositeType is not AWS
+    // Ideally we should refactor the model to include the compositeType using `for` attribute
+    if (compositeType !== TYPE_HOST_CATALOG_PLUGIN_AWS) {
+      delete serialized.worker_filter;
+    }
```

Comment:
Is this needed? You didn't include the `worker_filter` for the azureFields and it's deleted on line 93 for static host catalogs, don't those cover the scenarios for `workerFilter`?

Reply:
yeah this is needed, otherwise, it adds a `worker_filter : null` to Azure's json

Reply:
Ah I keep forgetting it's not an attribute